### PR TITLE
AG-684 Added gridPreDestroyed event

### PR DESCRIPTION
**Pull request by salmenus**

### CODE REVIEW

**Review by seanlandsman — COMMENTED** [review]
The vue part looks fine to me.

Do we not need an equivalent change in the vue3 and angular wrappers?

File: grid-community-modules/core/src/ts/gridOptionsService.ts [hunk 1]
```diff
@@ -73,6 +73,7 @@ export class GridOptionsService {
     // we store this locally, so we are not calling getScrollWidth() multiple times as it's an expensive operation
     private scrollbarWidth: number;
     private domDataKey = '__AG_' + Math.random().toString();
+    private static readonly alwaysSyncGlobalEvents: string[] = [Events.EVENT_GRID_PRE_DESTROYED];
```
Comment (StephenCooper) [inline grid-community-modules/core/src/ts/gridOptionsService.ts:76]: Can we make it a Set for performance reasons.

File: grid-community-modules/core/src/ts/gridOptionsService.ts [hunk 2]
```diff
@@ -205,17 +207,29 @@ export class GridOptionsService {
...
+    globalEventHandlerFactory = (restrictToSyncOnly?: boolean) => {
+        return (eventName: string, event?: any) => {
+            // prevent events from being fired _after_ the grid has been destroyed
+            if (this.destroyed) {
+                return;
+            }
+
+            if (
+                GridOptionsService.alwaysSyncGlobalEvents.includes(eventName) && !restrictToSyncOnly ||
+                !GridOptionsService.alwaysSyncGlobalEvents.includes(eventName) && restrictToSyncOnly
```
Comment (StephenCooper) [inline grid-community-modules/core/src/ts/gridOptionsService.ts:222]: Nit: Would prefer to reduce the duplication of the sync check via extracting to a nicely named variable.

File: grid-community-modules/core/src/ts/gridOptionsService.ts (hunk 1 above)
Comment (salmenus) [reply grid-community-modules/core/src/ts/gridOptionsService.ts:76]: Sure! it's a negligible gain given the size of the array, but let's do it for consistency with other implementations.

**Review by StephenCooper — APPROVED** [review]

**Pull request merged**